### DISCUSSION TITLE
jj noob: generating Linux kernel patch series with jj

### DISCUSSION THREAD

**Discussion by jagalactic** [jj-vcs/jj#3560] [General]
I recently discovered jj when I was facing a really nasty rebasing problem: turning a new Linux file system into a patch series where every patch/commit and makes functional sense, even though that task is impossible without stubbing some functionality in intermediate commits. Bravo - jj made the problem tractable, mostly via splitting and the ability to commit conflicts.

I fetched my jj branch back to a regular git tree, and immediately needed to rebase it to linux v6.9.0-rc4 (from rc3). Now I want to resume editing in jj, rebased to rc4, but I can't get new content into the jj tree (which was created by cloning from a local working tree with ```jj git clone linux linux-jj``` - a clone which takes forever on a linux kernel tree, BTW)

```
[jmg@jmg linux-jj]$ jj git remote list
origin /home/jmg/w/linux-debug
upstream //git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git
[jmg@jmg linux-jj]$ jj git fetch --remote origin
Error: object is not a committish; class=Invalid (3); code=Peel (-19)
```

There is a branch (as well as a tag) in my local origin that I would like to resume working on, but I'm not sure how to do that. I guess I can clone again, but that will take an hour or more. I also can't fetch from upstream. 

Thanks for any tips...

**Comment by joyously** [comment]
I am not the one to answer how to do it, but have you seen the [`sparse`](https://martinvonz.github.io/jj/prerelease/cli-reference/#jj-sparse) commands and/or [`workspace`](https://martinvonz.github.io/jj/prerelease/cli-reference/#jj-workspace)?

**Reply by jagalactic** [reply]
Thanks Joy; I knew about workspaces, but not sparse. I'm about to try workspaces since I'm working on more than one branch and that seems more practical for my particular flow than switching back & forth.

**Comment by martinvonz** [comment]
> ```
> [jmg@jmg linux-jj]$ jj git fetch --remote origin
> Error: object is not a committish; class=Invalid (3); code=Peel (-19)
> ```

Hmm, I wonder what it refers to that is "not a committish". Does `git --git-dir .jj/repo/store/git for-each-ref` return any refs that point to something that's not a commit (not `commit` in the second column)?

> a clone which takes forever on a linux kernel tree, BTW

We've noticed that libgit2 seems significantly slower than regular git to clone, unfortunately.

**Comment by jagalactic** [comment]
Thanks for your response Martin! The command above shows both commits and tags in the second column. It's a Linux kernel tree, so there are 10K refs:
```
commit: 6103
tag: 4941
```
I need to rebase my patch branch to a tag, so I might have intentionally tried to fetch tags.
